src/logger/log_entry.py

```diff
@@ -0,0 +1,166 @@
+from datetime import datetime
+
+import marshmallow as marsh
+from peewee import CharField, DateTimeField, ForeignKeyField, IntegerField
+
+from src.base_model import BaseModel
+from src.implementor.models import Run
+
+
+class LogEntry(BaseModel):
+    """Represents a single log entry. Used to log messages from the simulation."""
+
+    class Schema(BaseModel.Schema):
+        """The marshmallow schema for the LogEntry model."""
+
+        timestamp = marsh.fields.DateTime(required=True)
+        message = marsh.fields.String(required=True)
+        run_id = marsh.fields.UUID(required=True)
+
+        def _make(self, data: dict) -> "LogEntry":
+            return LogEntry(**data)
+
+    timestamp = DateTimeField(null=False, default=datetime.now())
+
+    message = CharField(null=False)
+    run_id = ForeignKeyField(Run, null=False)
```

Comment:
If you think it's possible that a run needs a reference to its log entries, please add a `backRef` 😇 
https://docs.peewee-orm.com/en/latest/peewee/relationships.html

Reply:
Please don't forget how "Abstract" classes will be represented as a table in the DB. Peewee will create for each class (abstract and subclasses) their own table. That means back referencing wouldn't work. If it's not needed, it's not a problem.

Reply:
I think, I do not want to use references for now. If I need them later, I will add them to all classes.